### ISSUE TITLE
Test cases implementation for SWIflow

### ISSUE THREAD

**Issue by etrujil** [USDA-ARS-NWRC/swiflow#7]
This is just a note on some of the work we need to add to the package.

We need to design a series of test cases for swiflow using different time series as input and test the output knowing what the mathematical result should be.

Guideline recommendations from Joachim Meyer should help with the structure:
https://softwareengineering.stackexchange.com/questions/260873/how-much-should-i-break-up-my-unit-tests

Ernesto

**Comment by scotthavens** [comment]
I would recommend running the original streamflow for a few different test cases, start with at least 2. We can use code coverage to ensure that everything is being tested and can add more tests to check for the corner cases.

**Comment by etrujil** [comment]
Yeah, we could create a square domain with a given SWI and run it through both, StreamFlow and SWIflow, after we do some basic tests.

**Comment by scotthavens** [comment]
There is a small RME domain in SMRF and AWSM that you could use. I'd recommend not square because we ran into a rotation problem in SMRF that was undetectable on a square domain...

**Comment by etrujil** [comment]
Well, rectangular, like all our domains. Would that have an issue? I am thinking something like a rectangular domain with 50-m cells and SWI of 1mm or something similar. A controlled input, with subwatershed dividing that domain, like quadrants.

**Comment by scotthavens** [comment]
The RME domain that we have would be a great test, it's rectangular and then you can use whatever inputs you desire at that point. I like the idea of a constant 1mm or even some random SWI over the domain?

**Comment by micahjohnson150** [comment]
It seems to me were talking about a couple tests.

1. Which I think should be first, is a repeatability test. Does our code as an ensemble still produce the same results, this is what the RME case is in SMRF/AWSM. 

2. Have we implemented this model the same as the SLF groups Streamflow module. Which should be on the docket as validation but I do see it as harder to do as I think we have abstracted the base theory and there may be other theory more associated to the temperature stuff that might create differences. 

3 (BONUS). A third test which may even be a combination of the first two is using the tollgate streamflow with a image of modeled swi.  Compare how close they are, then use that metric for reproducibility and as an assessment for our theory.

**Comment by etrujil** [comment]
I think the first should be one with a single time series of SWI as input that would simply test the functions. The ones above are already too complex as simple tests. For example:

- a unit time series at hourly intervals for one year (with a few sets of parameters)
- a seasonal (sine wave) type function with a period of one year, repeated multiple years, an a few parameter sets.

Then we move to spatial tests. Start simple, and add complexity as we clear these.
